### PR TITLE
[Kotlin Migration ] PreferenceExtensionsTest

### PR DESCRIPTION
**Pull request by Prince-kushwaha**
## Purpose / Description
Kotlin migration of PreferenceExtensionsTest
- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [x] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [x] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/test/java/com/ichi2/preferences/PreferenceExtensionsTest.kt [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    @Mock
+    private val mMockEditor: SharedPreferences.Editor? = null
+    private fun getOrSetString(key: String, supplier: Supplier<String?>): String? {
+        return getOrSetString(mMockPreferences!!, key, supplier)
+    }
+
+    @Before
+    fun setUp() {
+        MockitoAnnotations.openMocks(this)
+        Mockito.`when`(mMockPreferences!!.contains(VALID_KEY)).thenReturn(true)
```
Comment (Akshay0701) [inline AnkiDroid/src/test/java/com/ichi2/preferences/PreferenceExtensionsTest.kt:45]: you can import this, to reduce repetitive `Mockito.`

File: AnkiDroid/src/test/java/com/ichi2/preferences/PreferenceExtensionsTest.kt [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+import org.mockito.Mock
+import org.mockito.Mockito
+import org.mockito.MockitoAnnotations
+import java.lang.RuntimeException
+import java.util.function.Supplier
+
+// Unknown issue: @CheckResult should provide warnings on this class when return value is unused, but doesn't.
+class PreferenceExtensionsTest {
+    @Mock
+    private val mMockPreferences: SharedPreferences? = null
```
Comment (Akshay0701) [inline AnkiDroid/src/test/java/com/ichi2/preferences/PreferenceExtensionsTest.kt:34]: can be convert to lateinit

**Review by Akshay0701 — CHANGES_REQUESTED** [review]
Thanks for the contribution @Prince-kushwaha

File: AnkiDroid/src/test/java/com/ichi2/preferences/PreferenceExtensionsTest.kt (hunk 1 above)
Comment (Prince-kushwaha) [reply AnkiDroid/src/test/java/com/ichi2/preferences/PreferenceExtensionsTest.kt:45]: @Akshay0701 I did many kotlin Migration according to david-allison  kotlin code is same as it is in java  
No need to Import `Mockito.`

**Review by Akshay0701 — APPROVED** [review]

**Review by david-allison — APPROVED** [review]

**Pull request merged**